pkg/route/api/v1/conversion.go

```diff
@@ -13,6 +13,10 @@ func init() {
 			obj.To.Kind = "Service"
 		},
 		func(obj *TLSConfig) {
+			err := ValidateInsecureEdgeTerminationPolicy(obj)
```

Comment:
Existing routes that won't pass `ValidateInsecureEdgeTerminationPolicy` are already broken, right? It seems to me that this step isn't really necessary here.